src/Symfony/Component/DependencyInjection/Tests/ParameterBag/ParameterBagTest.php

```diff
@@ -82,32 +83,36 @@ public function testGetSet()
     }
 
     /**
-     * @group legacy
-     * Test it will throw in 7.0
+     * @dataProvider provideNumericNames
      */
-    public function testGetSetNumericName()
+    public function testSetNumericName(int|float $name)
     {
-        $bag = new ParameterBag(['foo']);
-        $bag->set(1001, 'foo');
-        $this->assertEquals('foo', $bag->get(1001), '->set() sets the value of a new parameter');
+        $bag = new ParameterBag();
 
-        $bag->set(10.0, 'foo');
-        $this->assertEquals('foo', $bag->get(10), '->set() sets the value of a new parameter');
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage(sprintf('The parameter name "%s" cannot be numeric.', $name));
 
-        $bag->set(0b0110, 'foo');
-        $this->assertEquals('foo', $bag->get(0b0110), '->set() sets the value of a new parameter');
+        $bag->set($name, 'foo');
+    }
 
-        $bag->set('0', 'baz');
-        $this->assertEquals('baz', $bag->get(0), '->set() overrides previously set parameter');
+    /**
+     * @dataProvider provideNumericNames
+     */
+    public function testConstructorNumericName(int|float $name)
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage(sprintf('The parameter name "%s" cannot be numeric.', $name));
 
-        $this->assertTrue($bag->has(0));
-        $this->assertTrue($bag->has(1001));
-        $this->assertTrue($bag->has(10));
-        $this->assertTrue($bag->has(0b0110));
+        new ParameterBag([$name => 'foo']);
+    }
 
-        foreach (array_keys($bag->all()) as $key) {
-            $this->assertIsInt($key, 'Numeric string keys are cast to integers');
-        }
+    public static function provideNumericNames(): \Generator
```

Comment:
Cannot be replaced by `@testWith` as `[0b0110]` is considered a syntax error by PHPUnit

Reply:
`0b0110` is just a number, we can drop this case (or replace it by `6`)

Reply:
I dropped the test case, it felt strange to me to keep a case that seems so close to the first one. I guess it doesn't bring that much after all :slightly_smiling_face: